src/components/UserInputComponent/UserInputForm.js

```diff
@@ -219,34 +217,56 @@ function UserInputForm(props) {
-                          let msg = null;
-                          Object.values(obj).forEach((err_msg) => {
-                            msg = (
-                              <span key={err_msg}>
-                                - {err_msg}
-                                <br />
-                              </span>
-                            );
-                          });
-                          return msg;
-                        })
-                      )}
-                      {selectedWayareaWithNoGDOGID && (
-                        <>- Valitulle väylälle ei löydy navigointilinjoja</>
+                      {/* When calculation type is not selected */}
+                      {selectedCalculationType == "" && (
+                        <>- Laskentatapa täytyy valita</>
                       )}
+                      {/* When calculation type is reittiviiva and reittiviiva is not selected*/}
+                      {selectedCalculationType == "reittiviiva" &&
+                        formik.values.routename == "" && <>- Valitse reitti</>}
+                      {/* When calculation type is navigationline and navigationline is no selected  */}
+                      {selectedCalculationType == "navigationline" &&
+                        (formik.values.vaylat == null ||
+                          formik.values.vaylat == "") && (
+                          <>- Valitse navigointilinja</>
+                        )}
+                      {/* When calculationtype is navigationline and navigationline is selected BUT there is no GDOGIDS for that navigation line */}
+                      {selectedCalculationType == "navigationline" &&
+                        formik.values.vaylat !== null &&
+                        formik.values.vaylat !== "" &&
+                        selectedWayareaWithNoGDOGID && (
+                          <>- Valitulle väylälle ei löydy navigointilinjoja</>
+                        )}
+                      {/* When calculation type is compare and either route name or reittiviiva is not selected */}
+                      {selectedCalculationType == "compare" &&
+                        (formik.values.routename == "" ||
+                          formik.values.vaylat == null ||
+                          formik.values.vaylat == "") && (
+                          <>- Valitse navigointilinja ja reitti</>
+                        )}
+                      {/* When calculationtype is compare and navigationline is selected BUT there is no GDOGIDS for that navigation line */}
+                      {selectedCalculationType == "compare" &&
+                        formik.values.vaylat !== null &&
+                        formik.values.vaylat !== "" &&
+                        selectedWayareaWithNoGDOGID && (
+                          <>- Valitulle väylälle ei löydy navigointilinjoja</>
+                        )}
                     </span>
                   </label>
                 ) : null
```

Comment:
The label validation could be refactored into its own functions, as it's quite large

Reply:
done